### PR TITLE
Ppd42 x

### PR DESCRIPTION
**Pull request by pi4homez**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>
**Pull request in [esphome-core](https://github.com/esphome/esphome-core) with C++ framework changes (if applicable):** esphome/esphome-core#<esphome-core PR number goes here>

## Checklist:

  - [X] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.

### CODE REVIEW

File: components/sensor/ppd42x.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    :width: 50.0%
+
+    PPD42X Particulate Matter Sensor
+Wiring:
+Pin Function    Wire color (in picture above)
+1   Ground	                   Green
+2   P2 (output)                White
+3   5V                         Yellow
+4   P1 (output)                Black
+5   Threshold voltage for P2   Red
```
Comment (OttoWinter) [inline components/sensor/ppd42x.rst]: This table won't show correctly. See waveshare_epaper as an example for these types of tables.

File: components/sensor/ppd42x.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    sensor:
+      - platform: ppd42x
+        time_out: 30000
+        pm_2_5:
+          name: "Particulate Matter <2.5µm Concentration"
+        pm_10_0:
+          name: "Particulate Matter <10.0µm Concentration"
+        update_interval: 5min
+
+With ``update_interval``, the working period of the PPD42X device will be changed. If ``update_interval`` is
```
Comment (OttoWinter) [inline components/sensor/ppd42x.rst:63]: Stale text, this is not true for ppd42x.

File: components/sensor/ppd42x.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+equal to ``0min``, the PPD42X will be set to continuous measurement and will report new measurement values
+approximately every second.
+
+If ``update_interval`` is between 1-30 minutes, the PPD42X periodically turns on for 30s before each measurement.
+For the remaining time the sensor is shut off. As a result, this mode can reduce power consumption and increases
+the lifetime of the PPD42X.
+
+Configuration variables:
+------------------------
+- **time_out** (*mandatory*, :ref:`config-time`): The interval to check the P1 and P2 pins in micro-seconds.
```
Comment (OttoWinter) [inline components/sensor/ppd42x.rst]: required not mandatory - and it *should* not be mandatory, it should be an optional variable and moved to the bottom.

File: components/sensor/ppd42x.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  This affects the working period of the PPD42X sensor. Defaults to ``30000 mico-seconds``.
```
Comment (OttoWinter) [inline components/sensor/ppd42x.rst]: If the user enters ``30000 mico-seconds`` validation will fail. Default value must be written in a way that passes validation

File: components/sensor/ppd42x.rst [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+- **pm_2_5** (*Optional*): Use the concentration of particulates of size less than 2.5µm in µg per cubic meter.
+  All options from :ref:`Sensor <config-sensor>`.
+
+- **pm_10_0** (*Optional*): Use the concentration of particulates of size less than 10.0µm in µg per cubic meter.
+  All options from :ref:`Sensor <config-sensor>`.
+
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor in minutes.
+  This affects the working period of the PPD42X sensor. Defaults to ``0min``.
```
Comment (OttoWinter) [inline components/sensor/ppd42x.rst]: Stale text plus incorrect default documented value

File: components/sensor/ppd42x.rst [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+.. note::
+
+    The configuration variable ``update_interval`` reconfigure the PPD42X device. This setting is still effective
+    after power off. This can affect the performance of other libraries. Factory default is continuous measurement.
+
+See Also
+--------
+
+- :doc:`/components/sensor/ppd42x`
```
Comment (OttoWinter) [inline components/sensor/ppd42x.rst]: Do not link the document you're currently viewing under see also, that makes no sense.

File: components/sensor/ppd42x.rst [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+    # Example configuration entry
+
+    sensor:
+      - platform: ppd42x
+        time_out: 30000
+        pm_2_5:
+          name: "Particulate Matter <2.5µm Concentration"
+        pm_10_0:
+          name: "Particulate Matter <10.0µm Concentration"
```
Comment (OttoWinter) [inline components/sensor/ppd42x.rst:59]: pin options are missing in this example here

**Review by OttoWinter — COMMENTED** [review]
- Move `ppd42x.jpg` to top-level `images/` folder
- Compress `ppd42x.jpg` with the command seen in the contributing docs
- Add ppd42x to the index.rst file so it's in the component index

File: components/sensor/ppd42x.rst (hunk 2 above)
Comment (pi4homez) [reply components/sensor/ppd42x.rst:63]: yep already corrected, i'm not the inventor of this device ;)

File: components/sensor/ppd42x.rst (hunk 3 above)
Comment (pi4homez) [reply components/sensor/ppd42x.rst]: ok , will change it in the code too

File: components/sensor/ppd42x.rst (hunk 4 above)
Comment (pi4homez) [reply components/sensor/ppd42x.rst]: got it

File: components/sensor/ppd42x.rst (hunk 7 above)
Comment (pi4homez) [reply components/sensor/ppd42x.rst:59]: yep

Comment (pi4homez) [reply components/sensor/ppd42x.rst:59]: done

**Pull request closed without merging**